operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -31,134 +30,79 @@
 import static oracle.kubernetes.operator.ProcessingConstants.DOMAIN_INTROSPECT_REQUESTED;
+import static oracle.kubernetes.operator.helpers.ConfigMapHelper.readExistingIntrospectorConfigMap;
 import static oracle.kubernetes.operator.logging.MessageKeys.INTROSPECTOR_JOB_FAILED;
 import static oracle.kubernetes.operator.logging.MessageKeys.INTROSPECTOR_JOB_FAILED_DETAIL;
 
 public class JobHelper {
 
-  static final String START_TIME = "WlsRetriever-startTime";
+  private static final int JOB_DELETE_TIMEOUT_SECONDS = 1;
   private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Operator", "Operator");
   static final String INTROSPECTOR_LOG_PREFIX = "Introspector Job Log: ";
   private static final String EOL_PATTERN = "\\r?\\n";
 
   private JobHelper() {
   }
 
-  static String createJobName(String domainUid) {
-    return LegalNames.toJobIntrospectorName(domainUid);
-  }
-
-  /**
-   * Factory for {@link Step} that creates WebLogic domain introspector job.
-   * Uses the following packet values:
-   *  ProcessingConstants.DOMAIN_TOPOLOGY - the domain topology
-   *  ProcessingConstants.DOMAIN_RESTART_VERSION - the restart version from the domain
-   *  ProcessingConstants.DOMAIN_INPUTS_HASH
-   *  ProcessingConstants.DOMAIN_INTROSPECT_VERSION - the introspect version from the old domain spec
-   *
-   * @param next Next processing step
-   * @return Step for creating job
-   */
-  public static Step createDomainIntrospectorJobStep(Step next) {
-
-    return new DomainIntrospectorJobStep(next);
-  }
-
-  private static boolean runIntrospector(Packet packet, DomainPresenceInfo info) {
-    WlsDomainConfig topology = (WlsDomainConfig) packet.get(ProcessingConstants.DOMAIN_TOPOLOGY);
-    LOGGER.fine("runIntrospector topology: " + topology);
-    LOGGER.fine("runningServersCount: " + runningServersCount(info));
-    LOGGER.fine("creatingServers: " + creatingServers(info));
-    LOGGER.fine("isModelInImageUpdate: " + isModelInImageUpdate(packet, info));
-    return topology == null
-          || isBringingUpNewDomain(packet, info)
-          || checkIfIntrospectionRequestedAndReset(packet)
-          || isModelInImageUpdate(packet, info)
-          || isIntrospectVersionChanged(packet, info);
-  }
-
-  private static boolean isBringingUpNewDomain(Packet packet, DomainPresenceInfo info) {
-    return runningServersCount(info) == 0 && creatingServers(info) && isGenerationChanged(packet, info);
-  }
-
-  private static boolean checkIfIntrospectionRequestedAndReset(Packet packet) {
-    return packet.remove(DOMAIN_INTROSPECT_REQUESTED) != null;
-  }
-
-  private static boolean isIntrospectVersionChanged(Packet packet, DomainPresenceInfo info) {
-    return Optional.ofNullable(packet.get(INTROSPECTION_STATE_LABEL))
-            .map(introspectVersionLabel -> !introspectVersionLabel.equals(getIntrospectVersion(info))).orElse(false);
-  }
+  //----------- for unit testing only ---------
```

Comment:
Is there a way to name these unit test methods so that it will be easier to recognize they are for unit test only?

Reply:
I'm open to suggestions, but most of them make specific reference to the UnitTestAdapter.

Reply:
I am concerned if someone trying to use these methods based on the names along later maybe confused until reading the actual code.  If the three methods cannot be encapsulated in UnitTestAdapter, maybe just rename with UnitTest appended to the end of the name.